packages/core/src/mm/actors/Player.S

```diff
@@ -1224,3 +1224,7817 @@ PATCH_END
 PATCH_START 0x8085e958
 .int 0
 PATCH_END
+
+/* Hammer Related ASM */
+
+PATCH_START 0x80123960
+  addiu     sp, sp, -0x20
+  sw        ra, 0x0014 (sp)
+  jal Player_CustomActionToModelGroup
+  nop
+  lw        ra, 0x0014 (sp)
+  jr        ra
+  addiu     sp, sp, 0x20
+PATCH_END
+
+PATCH_START 0x80128DDC
+  addiu     a0, s2, 0x0
+  jal       Player_GetCustomSwordLength
+  addiu     a1, s1, 0x0
+  j         0x80128E24
+  nop
+PATCH_END
+
+PATCH_START 0x808336EC
+  nop
+  addiu     a2, sp, 0x0018
+  jal       Player_SpecialMeleeWeaponAnim
+  sw	      v1, 0x0018 (sp)
+  lw        v1, 0x0018 (sp)
+  jal       0x801241B4
+PATCH_END
+
+PATCH_START 0x80833910
+jal       Player_GetMeleeAttackAnimInfo
+addiu     a2, s1, 0x0
+PATCH_END
+
+PATCH_START 0x808548CC
+  addiu     a1, a0, 0x0000
+  jal       Player_GetMeleeAttackAnimInfo
+  lb        a2, 0x0ada (s0)
+  addiu     t9, v0, 0x0000
+  lw        a1, 0x0044 (sp)
+PATCH_END
+
+PATCH_START 0x8082FAF4
+  beq       t9, at, .+0x0C
+  addiu     at, zero, PLAYER_CUSTOM_IA_HAMMER
+  beql      t9, at, .+0x04
+PATCH_END
+
+/* Hammer Related Relocs */
+
+PATCH_START 0x8085E9E8
+  .int 0x00000000
+PATCH_END
+
+PATCH_START 0x80860A18
+  .int 0x00000000
+PATCH_END
+
+PATCH_START 0x80860A74
+  .int 0x00000000
+PATCH_END
+
+PATCH_START 0x80861D88
+  .int 0x00000000
+  .int 0x00000000
+PATCH_END
+
+Player_BeginHammerShockwaveCheck:
+  lw	      a0, 0x0044 (sp)
+  sw        ra, 0x0024 (sp)
+  jal       Player_HammerShockwaveCheck
+  lw        a1, 0x0020 (sp)     
+  lw        ra, 0x0024 (sp)
+  jr ra
+  addiu     sp, sp, 0x40
+
+PATCH_START 0x80854C64
+  j Player_BeginHammerShockwaveCheck
+  nop
+PATCH_END 
+
+/* Hammer link_animetion Patches */
+/* These animations should overwrite unused Ganon/Ganondorf related animations from OoT */
+/* PATCH_VROM link_animation start + frame data offset */
+
+/* gPlayerAnim_Link_otituku_w -> gPlayerAnim_link_hammer_hit */
+PATCH_VROM 0x65d000 + 0xA45D0
```

Comment:
@Nax I hope you have a suggestion on how this can be improved.

Reply:
maybe these patches could be separated into a different file and `#include` could be used. I did this because there is no existing `link_animetion` file and it's not an object so there wasn't a folder for it either

Reply:
Please move this into a new file (maybe `Player_hammer_anims.S` or whatever) and add a `TODO` or `FIXME` comment. This will be okay for now but eventually I want to support cross-anims so we don't have to do that.